wazo_phoned/plugins/blf/services.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2020 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+
+from xivo import xivo_helpers
+
+logger = logging.getLogger(__name__)
+
+
+class BlfService:
+
+    DEVICE_PATTERN = 'Custom:{}'
+    INUSE = 'INUSE'
+    NOT_INUSE = 'NOT_INUSE'
+
+    def __init__(self, amid_client, confd_client):
+        self.amid = amid_client
+        self.confd = confd_client
+
+    def _device(self, user_uuid, name, destination=''):
+        funckey_prefix = self.confd.extensions_features.list(search='phoneprogfunckey')[
```

Comment:
I think we should do only one list here and filter here instead of doing 2 queries.
I also think the result should be cached and invalidated when extension features gets modified (I did not check if there an event that gets published)

Reply:
Done